File: web/src/shared/api/prints.ts
// Клиент набора принтов.

const BASE = import.meta.env.VITE_API_BASE ?? '/reference/api/'

export interface PrintItem {
  path: string
  name: string
  /** probe — эталон с вопросом, artwork — настоящий принт. */
  kind: 'probe' | 'artwork'
  subject: string | null
  answers: string | null
  /** Настоящий размер эталона на изделии, сантиметры. */
  width_cm: number | null
}

export async function fetchPrints(): Promise<PrintItem[]> {
  const r = await fetch(`${BASE}prints`)
  if (!r.ok) throw new Error(`Набор принтов: ${r.status}`)
  return (await r.json()) as PrintItem[]
}

export function printUrl(path: string): string {
  return `${BASE}prints/${path}`
}

File: api/reference_api/schemas/prints.py
"""Принты: что отдаётся наружу."""

from pydantic import BaseModel


class Print(BaseModel):
    path: str
    name: str
    # probe — эталон, отвечающий на вопрос про печать; artwork — настоящий принт.
    kind: str
    subject: str | None = None
    # На какой вопрос отвечает эталон. У настоящих принтов пусто.
    answers: str | None = None
    # Настоящий размер эталона на изделии. Сетка в 20 см проверяет
    # калибровку только если ложится ровно двадцатью сантиметрами.
    width_cm: float | None = None
